ws/config.py

```diff
@@ -1,180 +1,232 @@
 #! /usr/bin/env python3
 
+import argparse
+import configparser
+import logging
 import os
 import sys
-import collections
-import logging
-
-# TODO: make this an optional dependency, the from_argparser factory should work with plain argparse
-import ConfigArgParse.configargparse as configargparse
-import configfile
-# allow '-' in config keys and section names
-configfile.Section._OPTION = r'^[a-zA-Z_]+[a-zA-Z0-9_-]*$'
-configfile.Section._SECTION_SUB = r'^[a-zA-Z_]+(?:\.?[a-zA-Z0-9_-]+)*$'
-configfile.Section._SECTION_PLAIN = r'^[a-zA-Z_]+[a-zA-Z0-9_-]*$'
 
 import ws.logging
 
 logger = logging.getLogger(__name__)
 
-__all__ = ["ConfigFileParser", "argtype_existing_dir", "argtype_dirname_must_exist", "getArgParser", "object_from_argparser"]
+__all__ = [
+    "ConfigParser", 
+    "getArgParser", 
+    "object_from_argparser",
+    "argtype_bool",
+    "argtype_configfile",
+    "argtype_path",
+]
 
-class ConfigFileParser:
+PROJECT_NAME = 'wiki-scripts'
+CONFIG_DIR = '~/.config/'       # Can be overriden by XDG_CONFIG_HOME
+CACHE_DIR = '~/.cache/'         # Can be overriden by XDG_CACHE_HOME
+DEFAULT_CONF = 'default'        # Can be overriden with '--config' option
 
-    def __init__(self, top_level_arg, subname=None):
-        self.top_level_arg = top_level_arg
-        self.subname = subname
 
-    def parse(self, stream, context=None):
-        """
-        Parses a config file and returns a dictionary of settings
-        """
-        # TODO: convert (all?) exceptions to configargparse.ConfigFileParserException
-        cf = configfile.ConfigFile(stream, inherit_options=True, safe_calls=True, interpolation=True)
-
-        try:
-            _arg = "--" + self.top_level_arg
-            _i = context.index(_arg)
-            top_level = context[_i + 1]
-        except (ValueError, IndexError):
-            top_level = None
-
-        if top_level is None:
-            top_level = cf[self.top_level_arg]
-            if not top_level:
-                raise configargparse.ConfigFileParserException("top-level parameter '{}' not found")
-        return cf(top_level, self.subname).get_options()
-
-    def serialize(self, items):
-        """
-        Does the inverse of config parsing by taking parsed values and
-        converting them back to a string representing config file contents.
+# ==============
+# Parser classes
+# ==============
```

Comment:
Please remove this comment.

Reply:
I adopted this style in the official `argparse` module, but ok. Done.